code/modules/paperwork/fax.dm

```diff
@@ -65,14 +136,68 @@
 	return TOOL_ACT_TOOLTYPE_SUCCESS
 
 /obj/machinery/fax/attackby(obj/item/item, mob/user, params)
-	if(istype(item, /obj/item/paper))
-		if(!paper_contain)
-			paper_contain = item
+	if (jammed && clear_jam(item, user))
+		return
+	if (panel_open)
+		if (is_wire_tool(item))
+			wires.interact(user)
+		return
+	if (can_load_item(item))
+		if (!loaded_item_ref?.resolve())
+			loaded_item_ref = WEAKREF(item)
 			item.forceMove(src)
 			update_appearance()
 		return
 	return ..()
 
+/**
+ * Attempts to clean out a jammed machine using a passed item.
+ * Returns true if successful.
+ */
+/obj/machinery/fax/proc/clear_jam(obj/item/item, mob/user)
+	if (istype(item, /obj/item/reagent_containers/spray))
+		var/obj/item/reagent_containers/spray/clean_spray = item
+		if(!clean_spray.reagents.has_reagent(/datum/reagent/space_cleaner, clean_spray.amount_per_transfer_from_this))
+			return FALSE
+		clean_spray.reagents.remove_reagent(/datum/reagent/space_cleaner, clean_spray.amount_per_transfer_from_this, 1)
+		playsound(loc, 'sound/effects/spray3.ogg', 50, TRUE, MEDIUM_RANGE_SOUND_EXTRARANGE)
+		user.visible_message(span_notice("[user] cleans \the [src]."), span_notice("You clean \the [src]."))
+		jammed = FALSE
+		return TRUE
+	if (istype(item, /obj/item/soap) || istype(item, /obj/item/reagent_containers/cup/rag))
+		var/cleanspeed = 50
+		if (istype(item, /obj/item/soap))
+			var/obj/item/soap/used_soap = item
+			cleanspeed = used_soap.cleanspeed
+		user.visible_message(span_notice("[user] starts to clean \the [src]."), span_notice("You start to clean \the [src]..."))
+		if (do_after(user, cleanspeed, target = src))
+			user.visible_message(span_notice("[user] cleans \the [src]."), span_notice("You clean \the [src]."))
+			jammed = FALSE
+		return TRUE
+	return FALSE
+
+/**
+ * Returns true if an item can be loaded into the fax machine.
+ */
+/obj/machinery/fax/proc/can_load_item(obj/item/item)
+	if(!is_allowed_type(item))
+		return FALSE
+	if (!istype(item, /obj/item/stack))
+		return TRUE
+	var/obj/item/stack/stack_item = item
+	return stack_item.amount == 1
```

Comment:
Is this still needed? Do we want to prevent space cash transfering to 1 at a time?

Reply:
I think it both makes more sense and also is funnier to be limited to sending one bill at a time rather than an entire stack, but if you think it's too pedantic (or isn't worth the extra complication) I can cut it out.

Reply:
It's fine as is, was asking if you had a different opinion with materials gone.